libs/particles/effects.ts

```diff
@@ -28,6 +30,26 @@ namespace particles {
         return new ParticleEffect((anchor: ParticleAnchor, pps: number) => new ParticleSource(anchor, pps, factory));
     }
 
+    /**
+     * Removes all effects at anchor's location
+     * @param anchor the anchor to remove effects from
+     */
+    //% blockId=particlesremoveeffect block="remove effects on %anchor=variables_get(mySprite)"
+    //% group="Effects"
+    export function removeEffects(anchor: ParticleAnchor) {
+        const remainingEffects: ParticleSource[] = [];
+
+        ongoingEffects.forEach(ps => {
+            if (ps.anchor.x == anchor.x && ps.anchor.y == anchor.y) {
```

Comment:
shoulnd' you compare the reference?

Reply:
I was making it compare based off the anchor values so that things other than sprites work nicely; if it's based off of the reference matching then you wouldn't be able to turn off the effects here after they are created

![tmap](https://user-images.githubusercontent.com/5615930/50141091-7170ad00-025b-11e9-8bd4-bd2fd5cfaa7f.png)

Reply:
why not? can't you cycle through the tiles again and disable it?

Reply:
tiles aren't cached, so the array returned by ``array of all tiles`` will be new instances of the tiles

Reply:
comparing x/y is brittle

Reply:
I suppose just

```typescript
ps.anchor == anchor || ps.anchor.x == anchor.x && ps.anchor.y == anchor.y
```

can handle it a bit more clearly (I can't think of a case where it would be necessary though, except if the x and y getters were to have side effects)

Ideally I think it should still work even for something like

```
for (let i = 10; i < 160; i += 20) {
    particles.fireEffect.start({ x: i, y: screen.width / 2 }, i);
}

for (let i = 10; i < 160; i += 20) {
    pause(1000);
    particles.removeEffects({ x: i, y: screen.width / 2 });
}
```

though I suppose that's up for debate

Reply:
ref test makes it clearer so yes add it.